src/coreclr/nativeaot/Runtime/AsmOffsetsVerify.cpp

```diff
@@ -18,6 +17,9 @@
 #include "shash.h"
 #include <minipal/cpufeatures.h>
 
+#include "CommonMacros.inl"
```

Comment:
(Future cleanup: Including the .inl files manually like this is fragile. It would be best to make the .inl files included at the end of the .h files. It is the common pattern in C/C++.)

Reply:
The situation with includes is not the best. Everything seems to include a lot of random stuff - "slist.h"? Is everything fiddling with lists?  
I especially dislike that the order of includes often matters. I'd have points deducted in "C++ level 1" for that.

Rationalizing includes could be a tough knot to untangle.

Reply:
logged https://github.com/dotnet/runtime/issues/96081 about this.